src/inmanta/protocol/methods_v2.py

```diff
@@ -463,3 +463,41 @@ def resource_deploy_start(
     :return: A dict mapping the resource version id of each dependency of resource_id to
              the last deployment status of that resource.
     """
+
+
+# No pagination support is provided for this endpoint because there is no elegant way to page the output of this endpoint.
+@typedmethod(
+    path="/resource/<id>/events", operation="GET", arg_options=methods.ENV_OPTS, client_types=[ClientType.api], api_version=2
+)
+def get_resource_events(
+    tid: uuid.UUID,
+    id: model.ResourceVersionIdStr,
+) -> Dict[model.ResourceIdStr, List[model.ResourceAction]]:
+    """
+    Return relevant events for a resource, i.e. all deploy actions for each of its dependencies since this resources' last
+    deploy or all deploy actions if this resources hasn't been deployed before. The resource actions are sorted in descending
```

Comment:
all deploy actions can be a lot if history is deep. 

I wonder if there is another sane lower bound to the history? 
candidates 

- events since this resource is in the model 
- last N events 
- events since this resource has ti resource_attribute_hash
- events since time offset

Reply:
I think the `last N events` filter is the only one that puts an absolute upper bound on the number of entries returned by the query. Do I go for that approach?